modules/porous_flow/src/actions/PorousFlowUnsaturated.C

```diff
@@ -281,8 +288,9 @@ PorousFlowUnsaturated::addMaterials()
   if (_deps.dependsOn(_included_objects, "pressure_saturation_nodal"))
   {
     const std::string material_type = "PorousFlow1PhaseP";
-    InputParameters params = _factory.getValidParams(material_type);
     const std::string material_name = "PorousFlowUnsaturated_1PhaseP_VG_nodal";
+    InputParameters params = _factory.getValidParams(material_type);
+    params.set<std::vector<SubdomainName>>("block") = _subdomain_names;
```

Comment:
So because of the option to add subdomains during a simulation this is not exactly what we want. 

Basically the "no block restriction" should include the future subdomains that don't currently exist. 
And this line when it block restricts to everything excludes them.

Reply:
BTW: I got my "inspiration" over here: https://github.com/idaholab/moose/blob/a419bffe06d72af487b8a3d8bfd685689e9c69ed/modules/solid_mechanics/src/physics/QuasiStaticSolidMechanicsPhysics.C#L953
If this is not good due to the reasons you mention, this should be added of one of the todo-lists.

Now for some constructive remarks: 
- Do I unstanderstand correctly, that this line has to be adapted everywhere I introduced it? Not only the single line you are mentioning as an example?
- Do I understand correctly, that we should only do `params.set<std::vector<SubdomainName>>("block") = _subdomain_names;` in case the user provides a block-restriction explicitly?  
- Would an simple if-conditin do? Something like:
```
if (_subdomains_names_set_by_user)
  params.set<std::vector<SubdomainName>>("block") = _subdomain_names;
```

Reply:
```
if (isParamSetByUser("block"))
  params.set<std::vector<SubdomainName>>("block") = _subdomain_names;
```
should do it.

> If this is not good due to the reasons you mention,

it's not that it's not good, it's just that I know you are using this other feature and I don't want you to run into that problem

> Do I unstanderstand correctly, that this line has to be adapted everywhere I introduced it? Not only the single line you are mentioning as an example?

yes